### DISCUSSION TITLE
Multiple processes to one log file

### DISCUSSION THREAD

**Discussion by tsrich92** [odygrd/quill#164] [Q&A]
Can quill support logging from multiple processes into the same log file?

**Comment by diehard2** [comment]
I came here wondering the same thing and I didn't see anything while perusing the documentation. @odygrd is it officially supported?

**Comment by odygrd** [comment]
each process will spawn it's own backend logging thread, which will be trying to write into the same file. Quill doesn't do any file locking so that won't be possible.

It is possible however to implement your own `Handler` class (https://github.com/odygrd/quill/blob/master/examples/example_custom_handler.cpp) where you can implement file locking and then it should work and both processes will write to the file. However, the timestamps between the two processes might not be ordered.
